packages/server/graphql/public/mutations/requestToJoinDomain.ts

```diff
@@ -0,0 +1,42 @@
+import ms from 'ms'
+import {getUserId} from '../../../utils/authorization'
+import {MutationResolvers} from '../resolverTypes'
+import getKysely from '../../../postgres/getKysely'
+import isRequestToJoinDomainAllowed from '../../../utils/isRequestToJoinDomainAllowed'
+
+const REQUEST_EXPIRATION_DAYS = 30
+
+const requestToJoinDomain: MutationResolvers['requestToJoinDomain'] = async (
+  _source,
+  {},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const pg = getKysely()
+  const viewerId = getUserId(authToken)
+  const viewer = await dataLoader.get('users').loadNonNull(viewerId)
+  const domain = viewer.email.split('@')[1]
```

Comment:
+1 you can use https://github.com/ParabolInc/parabol/blob/9560f56fc97f09972d7a873157a2ce4206bd4450/packages/server/utils/getDomainFromEmail.ts instead